### ISSUE TITLE
Collection from modifier not found 3.3.1

### ISSUE THREAD

**Issue by wm-simon** [statamic/cms#5560]
### Bug description

Error after update to 3.3 with runtime parser. With regex it's working.

### How to reproduce

View:
```
{{ collection from="{collection}" limit="12" offset="5" sort="date:desc" as="links" }}
```

Error:
```
Statamic\Modifiers\ModifierNotFoundException
Modifier [from] not found
```

### Versions

Statamic 3.3.1 Pro
Laravel 8.83.5
PHP 8.0.12

### Installation

Fresh statamic/statamic site via CLI

**Comment by edalzell** [comment]
Try using a different variable name or using the binding syntax `:from=“collection”`

**Comment by edalzell** [comment]
Try using a different variable name or using the binding syntax `:from=“collection”`

**Comment by wm-simon** [comment]
`{{ collection :from="collection" limit="5" sort="date:desc" }}` causes the same error.
I can not use an other variable name, this would cause a lot of problems.

**Comment by JohnathonKoster** [comment]
I will be working on improvements to how situations like this are resolved (in #5545) - 

To force something to be a tag, you can prefix it with `%`:

```
{{ %collection from="{collection}" limit="5" sort="date:desc" }}

{{ /%collection }}
```

**Comment by wm-simon** [comment]
Hey @JohnathonKoster , thank you! That worked. 

But now I get an other error:
View:
`Statamic\Exceptions\TaxonomyNotFoundException
Taxonomy [] not found`
Error:
`<li class="item badge {{ taxonomy }} inline-block mb-1 ml-1 overflow-hidden !text-white rounded">`

Is this related or should I open a new issue?

**Comment by jackmcdade** [comment]
> But now I get an other error: View: `<li class="item badge {{ taxonomy }} inline-block mb-1 ml-1 overflow-hidden !text-white rounded">` Error: `Statamic\Exceptions\TaxonomyNotFoundException Taxonomy [] not found`

Unrelated – new issue it is!

**Comment by jasonvarga** [comment]
Using the exact snippet from the original issue:

```
{{ collection from="{collection}" limit="12" offset="5" sort="date:desc" as="links" }}
```

This works fine for me. There must be something else in your template that would be affecting this. Could you provide any more details?

**Comment by wm-simon** [comment]
I call the collection tag inside a partial and give the collection name as variable:
```
// Parent View
{{ partial:modules/home/new bg-color="bg-pri-100" title="Gastronomie" collection="gastronomie" }}  

// Partial
{{ collection from="{collection}" limit="12" offset="5" sort="date:desc" as="links" }}
```
I think the error is caused, when you have a variable called `collection` or `taxonomy`, like #5570 .

**Comment by jasonvarga** [comment]
This appears to be fixed by #5571 

The issue was only reproducible because you were passing a `collection` string into a partial.
